packages/react/src/ActionList/Visuals.tsx

```diff
@@ -77,3 +80,56 @@ export const TrailingVisual: React.FC<React.PropsWithChildren<VisualProps>> = ({
     </Box>
   )
 }
+
+// VisualOrIndicator handles the positioning of indicators and determines whether to show a visual or an indicator.
+//
+// If we're showing an *inactive* or *loading* indicator and a leading visual has NOT been passed,
+// replace the trailing visual with the inactive indicator.
+//
+// This preserves the left alignment of item text.
+export const VisualOrIndicator: React.FC<
+  React.PropsWithChildren<{
+    inactiveText?: TooltipProps['text']
+    itemHasLeadingVisual: boolean
+    labelId?: string
+    loading?: boolean
+    position: 'leading' | 'trailing'
+  }>
+> = ({children, labelId, loading, inactiveText, itemHasLeadingVisual, position}) => {
+  const VisualComponent = position === 'leading' ? LeadingVisual : TrailingVisual
+
+  if (!loading && !inactiveText) return children
+
+  if (
+    (itemHasLeadingVisual && position === 'trailing') || // has a leading visual, and it's in the trailing position, or
+    (!itemHasLeadingVisual && position === 'leading') // it doesn't have a leading visual, and it's in the leading position
+  ) {
+    // => so we don't render the indicator here
+    return children
+  }
+
+  return inactiveText ? (
+    <Tooltip text={inactiveText} type="label">
+      <Box
+        as="button"
+        sx={{
+          background: 'none',
+          color: 'inherit',
+          border: 'none',
+          padding: 0,
+          font: 'inherit',
+          cursor: 'pointer',
+        }}
+        aria-describedby={labelId}
+      >
+        <VisualComponent>
+          <AlertIcon />
+        </VisualComponent>
+      </Box>
+    </Tooltip>
+  ) : (
+    <VisualComponent>
+      <Spinner size="small" />
```

Comment:
I see that `<Spinner>` now provides the "loading" text, but `ActionList.Item` isn't referencing it 😭 This is an issue with `ActionList.Item` utilizing `aria-labelledby` and referencing only the container for the visual text.

I think we have a few options we could take:
* Add `<VisuallyHidden>Loading</VisuallyHidden>` in `Item.tsx` that is only present when in loading state
* Add generated ID to `VisualOrIndicator` that is only used when in loading state that [we add to the `aria-labelledby`](https://github.com/primer/react/blob/main/packages/react/src/ActionList/Item.tsx#L335)

Reply:
I went with the `<VisuallyHidden>` direction